sstables/sstables.cc

```diff
@@ -1618,6 +1618,7 @@ sstable::write_scylla_metadata(const io_priority_class& pc, shard_id shard, ssta
     _components->scylla_metadata->data.set<scylla_metadata_type::Sharding>(std::move(sm));
     _components->scylla_metadata->data.set<scylla_metadata_type::Features>(std::move(features));
     _components->scylla_metadata->data.set<scylla_metadata_type::RunIdentifier>(std::move(identifier));
+    _components->scylla_metadata->data.set<scylla_metadata_type::CannotHavePartitionTombstones>(std::move(cannot_have_partition_tombstones));
```

Comment:
I find the name "cannot have" a little awkward. It's describing something already done, and immutable. More natural to say "has no".

Reply:
Or even, "has partition tombstones", to avoid the double negative.

No entry -> unknown
true -> has partition tombstones
false -> does not have partition tombstones

Reply:
but the negation to "has no" is "has", which would suggest that the flag being false means that the sstable has partition tombstones. But it merely means that it may have - we don't know if it does.

Reply:
https://github.com/scylladb/scylla/pull/7437#discussion_r521200478

Reply:
Ok, maybe I'll just use an enum class `HasPartitionTombstones` with three options: `Yes`, `No`, `Unknown`

Reply:
There is a payload, a bool, no?

So if the section is missing, it's unknown, and if it exists, it's known, and the bool indicates what.

Reply:
I was worried about the API. If we define a function `has_partition_tombstones()`, it shouldn't return a `bool`, it should return a type with 3 possible values. On the other hand, if we define a function like `can_have_partition_tombstones()` or `cannot_have_partition_tombstones()` or `may_have_partition_tombstones()` or whatever, then it can return a `bool` and the semantics are consistent with the naming.

Reply:
check out the new version, now using enum class with 3 possible values.